Simple-Projects/Simple_Projects/Car/Car.cpp

```diff
@@ -47,18 +47,19 @@ void Car::drive(double x, double y) {
 }
 
 void Car::refill(double fuel) {
-    double newFuelAmount = fuelAmount + fuel;
-    fuelAmount = newFuelAmount;
+    const double newFuelAmount = fuelAmount + fuel;
 
     if ( newFuelAmount > fuelCapacity ) {
-        throw ToMuchFuel();
+        throw TooMuchFuel();
     }
+
+    fuelAmount += newFuelAmount;
 }
 
 std::ostream& operator<<(std::ostream& out, const Car& car) {
-    out << "Car: " << car.getModel() << std::endl;
-    out << "Car Location: " << car.getLocation() << std::endl;
-    out << "Fuel Amount: " << car.getFuelAmount() << std::endl;
-    out << "Consumption: " << car.getFuelConsumption() << std::endl;
+    out << "Car: " << car.getModel() << '\n';
```

Comment:
endl flushes the stream buffer. It is not needed so frequently - let the caller decide on when in needs to be done for responsive I/O